services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +45,48 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+  if (thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_ARG;
+  if (xQueueSend(thermalMgrQueueHandle,event,0) == errQUEUE_FULL) return ERR_CODE_QUEUE_FULL;//never block bc this function is called in an ISR
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type=THERMAL_MGR_EVENT_INTERRUPT;
```

Comment:
This ensures the rest of the struct gets initialized to zero (if the struct held more than just `type`).
```suggestion
  thermal_mgr_event_t event = {.type = THERMAL_MGR_EVENT_INTERRUPT};
```